tenders/hvt/hvt_module_blk.c

```diff
@@ -108,30 +108,62 @@ static void hypercall_block_read(struct hvt *hvt, hvt_gpa_t gpa)
 
 static int handle_cmdarg(char *cmdarg, struct mft *mft)
 {
-    if (strncmp("--block:", cmdarg, 8) != 0)
+    enum {
+        opt_block,
+        opt_block_size,
+    } which;
+    if (strncmp("--block:", cmdarg, 8) == 0)
```

Comment:
Size of constant strings can be computed at compile time:
```c
#define STARTS_WITH(prefix, str) (strncmp(prefix, cmdarg, sizeof(prefix) - 1) == 0)
```

```c
    if (STARTS_WITH("--block:", cmdarg))
```

Perhaps a cast can be used to annotate the type of `prefix` in the macro ? eg. `sizeof((char[])prefix)`, not sure if that would give the right result.

Reply:
Hmm, yes, that's a good point. It's basically the same code as in `*_module_net`. I don't think we can use a cast to assert the type. Since we use the same pattern in (at least) 4 different files I wonder where it would make sense to define the macro.

Reply:
`tenders/common/cc.h` might have room for such macro.

Reply:
This macro doesn't misbehave but is weirder:
```c
#define STARTS_WITH(prefix, str) (strncmp(#prefix, str, sizeof(#prefix) - 1) == 0)

STARTS_WITH(--block:, cmdarg)
```
But this might be better defined as a inlineable C function. `strlen` might be evaluated at compile time but that doesn't matter anyway.